torch/_higher_order_ops/wrap.py

```diff
@@ -9,5 +10,23 @@ def __call__(self, func, *args):
         result = func(*args)
         return result
 
-
 wrap = Wrap()
+
+class WrapActivationCheckpoint(HigherOrderOperator):
+    """
+    Wrap activation checkpoint is little different from Wrap. There are two
+    functions to wrap here - utils.checkpoint() itself and the function to be
+    checkpointed (the first arg to the utils.checkpoint() function).
+    """
+    def __init__(self):
+        super().__init__("wrap_activation_checkpoint")
+
+    def __call__(self, function, *args, **kwargs):
+        # use_reentrant is set to False because this op is going to be traced.
+        # And we ensure that AOT Autograd traces through the non reentrant
+        # version of checkpointing.
+        kwargs["use_reentrant"] = False
```

Comment:
One other way of doing it is that if use_reentrant=True, then we graph break. IDK what's better here.

Reply:
Will keep use_reentrant=False. Idea is that we will mostly use checkpoint higher order op for tracing. And in that case, it does not matter to the user if we are tracing entrant vs non-reentrant. We will support non-reentrant version with AOT Autograd tracing.